File: БР1.2/Smirnova_Karina/labs/lab2/src/main/java/org/renting/rentingservice/mapper/BookingMapper.java
package org.renting.rentingservice.mapper;

import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.renting.rentingservice.domain.entity.BookingEntity;
import org.renting.rentingservice.dto.booking.BookingResponse;

@Mapper(componentModel = "spring")
public interface BookingMapper {

    @Mapping(target = "listingId", source = "listing.id")
    @Mapping(target = "guestId", source = "guest.id")
    BookingResponse toResponse(BookingEntity entity);
}

File: БР1.2/Цатинян Артём/labs/lab1/src/main/kotlin/ru/itmo/restaurantbooking/booking/adapter/rest/mapper/BookingMapper.kt
package ru.itmo.restaurantbooking.booking.adapter.rest.mapper

import org.mapstruct.Mapper
import org.mapstruct.Mapping
import org.mapstruct.MappingConstants
import ru.itmo.restaurantbooking.booking.adapter.rest.dto.BookingRestaurantResponse
import ru.itmo.restaurantbooking.booking.adapter.rest.dto.BookingSearchRequest
import ru.itmo.restaurantbooking.booking.adapter.rest.dto.BookingTableResponse
import ru.itmo.restaurantbooking.booking.service.BookingSearchQuery
import ru.itmo.restaurantbooking.jooq.tables.pojos.RestaurantTables
import ru.itmo.restaurantbooking.jooq.tables.pojos.Restaurants

@Mapper(componentModel = MappingConstants.ComponentModel.SPRING)
interface BookingMapper {
    @Mapping(source = "sortDir", target = "sortDirection")
    fun toSearchQuery(request: BookingSearchRequest): BookingSearchQuery

    fun toRestaurantResponse(restaurant: Restaurants): BookingRestaurantResponse

    fun toTableResponse(table: RestaurantTables): BookingTableResponse
}
